CRM/Core/DAO.php

```diff
@@ -186,16 +186,6 @@ public static function init($dsn) {
     }
     $factory = new CRM_Contact_DAO_Factory();
     CRM_Core_DAO::setFactory($factory);
-    $currentModes = CRM_Utils_SQL::getSqlModes();
-    if (CRM_Utils_Constant::value('CIVICRM_MYSQL_STRICT', CRM_Utils_System::isDevelopment())) {
-      if (CRM_Utils_SQL::supportsFullGroupBy() && !in_array('ONLY_FULL_GROUP_BY', $currentModes) && CRM_Utils_SQL::isGroupByModeInDefault()) {
```

Comment:
I guess I never looked at what supportsFullGroupBy() does:

```
    if (stripos($version, 'mariadb') !== FALSE) {
      return FALSE;
    }
```

That explains a lot now. Anyway it doesn't change anything about the PR.

Reply:
Yeh MySQL 5.7 and later's understanding of the ONLY_FULL_GROUP_BY sql mode is different to MariaDB's and MariaDB's is way more stricter

Reply:
https://mysqlserverteam.com/mysql-5-7-only_full_group_by-improved-recognizing-functional-dependencies-enabled-by-default/